### PR TITLE
Fix dockerfile goflags

### PR DESCRIPTION
**Pull request by mikenairn**
Fixes an issue where the operator-sdk commands would fail with:

```
go test: mod flag may be set only once
run "go help test" or "go help testflag" for more information
Error: failed to build test binary: (failed to exec []string{"go", "test", "-mod=vendor", "./openshift-ci/test/e2e/...", "-namespacedMan", "/tmp/namespaced-manifest.yaml242874436", "-globalMan", "/tmp/global-manifest.yaml240851667", "-root", "/integreatly-operator", "-timeout=60m", "-singleNamespace", "-parallel=1"}: exit status 2)
```

Also fixes the vendor drectory which appears to be out of date.

### CODE REVIEW

File: openshift-ci/Dockerfile.tools [hunk 1]
```diff
@@ -1,6 +1,7 @@
 FROM registry.svc.ci.openshift.org/openshift/release:golang-1.13
 
-ENV OPERATOR_SDK_VERSION=v0.12.0
+ENV OPERATOR_SDK_VERSION=v0.12.0 \
+    GOFLAGS=""
```
Comment (mikenairn) [inline openshift-ci/Dockerfile.tools]: @psturc @laurafitzgerald We should make this change in heimdall and cro also

File: openshift-ci/test/run [hunk 2]
```diff
@@ -15,5 +15,9 @@ docker run --rm -u $UID -v "${WORK_DIR}/../..:/integreatly-operator:z" --entrypo
 docker run --rm -u $UID -v "${WORK_DIR}/../..:/integreatly-operator:z" --entrypoint=/bin/sh ${IMAGE_NAME} -c "cd /integreatly-operator && make code/compile"
 docker run --rm -u $UID -v "${WORK_DIR}/../..:/integreatly-operator:z" --entrypoint=/bin/sh ${IMAGE_NAME} -c "cd /integreatly-operator && make test/unit"
 
+#Requires you to be oc logged in to any cluster on your host machine
+INTEGREATLY_OPERATOR_IMAGE="registry.svc.ci.openshift.org/integr8ly/integreatly-operator"
+E2E_CMD="operator-sdk --verbose test local ./openshift-ci/test/e2e --namespace=test --go-test-flags \"-timeout=60m\" --debug --image=${INTEGREATLY_OPERATOR_IMAGE}"
+docker run --rm -e KUBECONFIG=/kube.config -v "${HOME}/.kube/config":/kube.config:z -u $UID -v "${WORK_DIR}/../..:/integreatly-operator:z" --entrypoint=/bin/sh ${IMAGE_NAME} -c "cd /integreatly-operator && ${E2E_CMD}"
```
Comment (mikenairn) [inline openshift-ci/test/run]: Simple test to make sure that the operator-sdk will actually execute inside the container we are creating.

Without changes to Dockerfile in this PR:

```
$ IMAGE_NAME=registry.svc.ci.openshift.org/integr8ly/intly-operator-base-image:old test/run
.....
time="2020-01-15T18:42:40Z" level=debug msg="Debug logging is set"                                                                                                                                                 
time="2020-01-15T18:42:40Z" level=info msg="Testing operator locally."                                   
time="2020-01-15T18:42:40Z" level=debug msg="Running []string{\"go\", \"test\", \"-mod=vendor\", \"./openshift-ci/test/e2e/...\", \"-namespacedMan\", \"/tmp/namespaced-manifest.yaml242874436\", \"-globalMan\", \
"/tmp/global-manifest.yaml240851667\", \"-root\", \"/integreatly-operator\", \"-timeout=60m\", \"-singleNamespace\", \"-parallel=1\"}"
go test: mod flag may be set only once                                                                   
run "go help test" or "go help testflag" for more information                                            
Error: failed to build test binary: (failed to exec []string{"go", "test", "-mod=vendor", "./openshift-ci/test/e2e/...", "-namespacedMan", "/tmp/namespaced-manifest.yaml242874436", "-globalMan", "/tmp/global-man
ifest.yaml240851667", "-root", "/integreatly-operator", "-timeout=60m", "-singleNamespace", "-parallel=1"}: exit status 2)
Usage:                                                                                                   
  operator-sdk test local <path to tests directory> [flags] 
```

With the Dockerfile changes in this PR:

```
$ IMAGE_NAME=registry.svc.ci.openshift.org/integr8ly/intly-operator-base-image:latest test/run
....
time="2020-01-15T18:53:38Z" level=debug msg="Debug logging is set"
time="2020-01-15T18:53:38Z" level=info msg="Testing operator locally."
time="2020-01-15T18:53:38Z" level=debug msg="Running []string{\"go\", \"test\", \"-mod=vendor\", \"./openshift-ci/test/e2e/...\", \"-namespacedMan\", \"/tmp/namespaced-manifest.yaml953664498\", \"-globalMan\", \"/tmp/global-manifest.yaml809173161\", \"-root\", \"/integreatly-operator\", \"-timeout=60m\", \"-singleNamespace\", \"-parallel=1\"}"
testing: warning: no tests to run
PASS
ok      github.com/integr8ly/integreatly-operator/openshift-ci/test/e2e 1.456s [no tests to run]
time="2020-01-15T18:54:10Z" level=info msg="Local operator test successfully completed."
SUCCESS!
```

File: openshift-ci/test/e2e/main_test.go [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package e2e
```
Comment (laurafitzgerald) [inline openshift-ci/test/e2e/main_test.go]: I think this test directory should be in the root of the repo rather than under `openshift-ci`.

Comment (mikenairn) [reply openshift-ci/test/e2e/main_test.go]: Only just saw this. It's here for a reason, it's not a real test for the operator, it's just one for the Dockerfile itself.

**Pull request merged**